core/src/main/java/jenkins/util/AntClassLoader.java

```diff
@@ -919,6 +911,26 @@ public URL getResource(String name) {
         return url;
     }
 
+    /**
+     * Finds a matching file by iterating pathComponents.
+     *
+     * @param pathComponents Path to a folder, split into the individual folder names
+     * @param name File to find
+     * @return Url to found object
+     */
+    @CheckForNull
+    protected URL getUrl(ArrayList<File> pathComponents, String name) {
```

Comment:
```suggestion
    protected URL getUrl(Iterable<File> pathComponents, String name) {
```

Just to make it more generic